components/fxa-client/ios/FxAClient/FxAccountManager.swift

```diff
@@ -440,6 +447,24 @@ open class FxAccountManager {
 
                 return Event.fetchProfile
             }
+            case let .changedPassword(newSessionToken): do {
+                do {
+                    try requireAccount().handleSessionTokenChange(sessionToken: newSessionToken)
+
+                    FxALog.info("Initializing device")
+                    requireConstellation().initDevice(
+                        name: deviceConfig.name,
+                        type: deviceConfig.type,
+                        capabilities: deviceConfig.capabilities
+                    )
```

Comment:
It's a little sad that `handleSessionChange` can't just handle the device re-registration internally, but NBD.